service/header/subscription.go

```diff
@@ -0,0 +1,56 @@
+package header
+
+import (
+	"context"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+// ExtendedHeaderSubTopic hardcodes the name of the ExtendedHeader
+// gossipsub topic.
+const ExtendedHeaderSubTopic = "header-sub"
+
+// subscription handles retrieving ExtendedHeaders from the header pubsub topic.
+type subscription struct {
+	topic        *pubsub.Topic
+	subscription *pubsub.Subscription
+}
+
+// newSubscription creates a new ExtendedHeader event subscription
+// on the given host.
+func newSubscription(topic *pubsub.Topic) (*subscription, error) {
+	sub, err := topic.Subscribe()
+	if err != nil {
+		return nil, err
+	}
+
+	return &subscription{
+		topic:        topic,
+		subscription: sub,
+	}, nil
+}
+
+// NextHeader returns the next (latest) ExtendedHeader from the network.
+func (s *subscription) NextHeader(ctx context.Context) (*ExtendedHeader, error) {
+	msg, err := s.subscription.Next(ctx)
```

Comment:
Q: Is the header validated at this point already validated or validated to some extent? If not, where does the validation happen?

Reply:
I think it gets validated via pubsub validator (have no idea how this works) but I defer to @Wondertan

Reply:
The header here is already validated. By design, not yet by the implementation(#157). We will add validation for the topic using pubsub internal mechanism for msg verification.

Reply:
To extend here, a Header here is validated for a valid commit, but not fully validated by DASing.

Reply:
Resolving as it seems like @liamsi saw the comment.